osmdata.cpp

```diff
@@ -82,15 +84,21 @@ int osmdata_t::node_modify(osmium::Node const &node)
 {
     slim_middle_t *slim = dynamic_cast<slim_middle_t *>(mid.get());
 
-    slim->nodes_delete(node.id());
-    slim->nodes_set(node);
+    osmium::Location oldnode = slim->nodes_get(node.id());
+    if ( node.location().x() != oldnode.x() || node.location().y() != oldnode.y() ) {
```

Comment:
osmium::Location has both == and != operators, so `node.location() != oldnode` should do here.

 https://github.com/osmcode/libosmium/blob/master/include/osmium/osm/location.hpp#L489